src/tequila/apps/krylov/krylov.py

```diff
@@ -0,0 +1,47 @@
+import copy
+import scipy
+import tequila as tq
+from tequila import braket
+from tequila.hamiltonian.qubit_hamiltonian import QubitHamiltonian
+
+
+def krylov_method(krylov_circs:list, H:QubitHamiltonian, variables:dict=None)->tuple:
+    """Function that applies Krylov method to an Hamiltonian operator,
+    given the list of Krylov quantum circuits. If the circuits are parametrized 
+    also the variables need to be passed. The method returns the ground state energy 
+    and the array of coefficients allowing to obtain an approximation of the ground state.
+
+    Args:
+        krylov_circs (list): List of Krylov circuits.
+        H (QubitHamiltonian): Hamiltonian on which we want to apply Krylov method
+        variables (dict, optional): Dicitionary containing possible variables to be stored in the Krylov circuits. 
+        Defaults to None.
+
+    Returns:
+        tuple(float, np.ndarray): ground state energy, krylov coefficients
+    """
+    
+    n_krylov_states = len(krylov_circs)
+    HM = tq.QTensor(shape=[n_krylov_states,n_krylov_states])
+    SM = tq.QTensor(shape=[n_krylov_states,n_krylov_states])
+    
+    if variables != None:
```

Comment:
flake8 will complain about this later when I merge, can you change to: 
if variables is not None